cloudformation/dashboard.txt

```diff
@@ -0,0 +1,51 @@
+{
```

Comment:
How about making this a `.js` file? Might be easier to maintain due to syntax highlighting and eslint?

```js
'use strict';

module.exports.dashboard = () =>
  JSON.stringify({
    widgets: [
      {
        type: 'text',
        x: 0,
        y: 0,
        width: 24,
        height: 1,
        properties: {
          markdown: '## ${self:service}',
        }.
      },
      {
        type: 'metric',
        x: 0,
        y: 1,
        width: 6,
        height: 3,
        properties: {
          view: 'singleValue',
          metrics: [
            [
              '${self:service}',
              'BuyPrice',
              'CryptoCurrency',
              '${self:provider.environment.PREFERRED_CRYPTO_CURRENCY}',
              'Stage',
              '${self:custom.stage}',
              'LocalCurrency',
              '${self:provider.environment.PREFERRED_LOCAL_CURRENCY}',
              {
                label: 'Buy',
              },
            ],
            [
              '.',
              'SellPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                label: 'Sell',
              },
            ],
          ],
          region: '${self:provider.region}',
          title: 'Current price in ${self:provider.environment.PREFERRED_LOCAL_CURRENCY}',
        },
      },
      {
        type: 'metric',
        x: 0,
        y: 4,
        width: 24,
        height: 9,
        properties: {
          view: 'timeSeries',
          stacked: false,
          metrics: [
            [
              '${self:service}',
              'BuyPrice',
              'CryptoCurrency',
              '${self:provider.environment.PREFERRED_CRYPTO_CURRENCY}',
              'Stage',
              '${self:custom.stage}',
              'LocalCurrency',
              '${self:provider.environment.PREFERRED_LOCAL_CURRENCY}',
              {
                color: '#aec7e8',
                label: 'Buy actual',
              },
            ],
            [
              '.',
              'SellPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                color: '#ffbb78',
                label: 'Sell actual',
              },
            ],
            [
              '.',
              'BuyPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                period: 3600,
                color: '#1f77b4',
                label: 'Buy hourly average',
              },
            ],
            [
              '.',
              'SellPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                period: 3600,
                color: '#ff7f0e',
                label: 'Sell hourly average',
              },
            ],
            [
              '.',
              'BuyPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                period: 86400,
                label: 'Buy daily average',
                color: '#ff9896',
              },
            ],
            [
              '.',
              'SellPrice',
              '.',
              '.',
              '.',
              '.',
              '.',
              '.',
              {
                period: 86400,
                label: 'Sell daily average',
                color: '#98df8a',
              },
            ],
          ],
          region: '${self:provider.region}',
          title: 'Buy / Sell ${self:provider.environment.PREFERRED_CRYPTO_CURRENCY}-${self:provider.environment.PREFERRED_LOCAL_CURRENCY}',
        },
      },
    ],
  });
```

Refferencing it like:
```yml
DashboardBody : ${file(cloudformation/dashboard.js):dashboard}
```

☝️ Not sure if this would still parse those `${}` strings afterwards inside Serverless. If it doesn't we could add the following to the `.js` file.

```js
const config = require('../package.json');

// replace all ${self:service} references with:
{
...
  markdown: `## ${config.name}`,
...
}

// replace all ${self:provider.environment} references with:
{
...
  foo: `${process.env.PREFERRED_CRYPTO_CURRENCY}`,
...
}
```